sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentWriteResourceIntegrationTest.java

```diff
@@ -0,0 +1,388 @@
+  public static final String DEFAULT_NAMESPACE = RandomStringUtils.randomAlphanumeric(16);
+  public static final String DEFAULT_COLLECTION = RandomStringUtils.randomAlphanumeric(16);
+  public static final String DEFAULT_PAYLOAD =
+      "{\"test\": \"document\", \"this\": [\"is\", 1, true]}";
+  public static final String MALFORMED_PAYLOAD = "{\"malformed\": ";
+
+  @Inject NamespaceManager namespaceManager;
+
+  @Inject TableManager tableManager;
+
+  @Inject ObjectMapper objectMapper;
+
+  @BeforeAll
+  public void init() {
+
+    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+
+    namespaceManager
+        .createNamespace(DEFAULT_NAMESPACE, Replication.simpleStrategy(1))
+        .await()
+        .atMost(Duration.ofSeconds(10));
+
+    tableManager
+        .createCollectionTable(DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+        .await()
+        .atMost(Duration.ofSeconds(10));
+  }
+
+  @Nested
+  class WriteDocument {
+    @Test
+    public void happyPath() {
+      Response postResponse =
+          given()
+              .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+              .header("Content-Type", "application/json")
+              .body(DEFAULT_PAYLOAD)
+              .when()
+              .post(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+              .peek();
+
+      assertThat(postResponse.statusCode()).isEqualTo(201);
+      assertThat(postResponse.header("location")).isNotNull();
+      String location = postResponse.header("location");
+
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .queryParam("raw", "true")
+          .when()
+          .get(location)
+          .then()
+          .statusCode(200)
+          .body(jsonEquals(DEFAULT_PAYLOAD));
+    }
+
+    @Test
+    public void withProfile() {
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .header("Content-Type", "application/json")
+          .queryParam("profile", "true")
+          .body(DEFAULT_PAYLOAD)
+          .when()
+          .post(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(201)
+          .body("profile", notNullValue());
+    }
+
+    @Test
+    public void withTtl() {
+      Response postResponse =
+          given()
+              .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+              .header("Content-Type", "application/json")
+              .queryParam("ttl", "1")
+              .body(DEFAULT_PAYLOAD)
+              .when()
+              .post(BASE_PATH, DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+              .peek();
+
+      assertThat(postResponse.statusCode()).isEqualTo(201);
+      assertThat(postResponse.header("location")).isNotNull();
+      String location = postResponse.header("location");
+
+      Awaitility.await()
+          .atLeast(0, TimeUnit.MILLISECONDS)
```

Comment:
`atLeast` is not used correct here (https://github.com/awaitility/awaitility/wiki/Usage#at-least), actually has no affect.. Remove.. 

`atMost` with 2 seconds (1s ttl +1s buffer) seems like the correct way imo